hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/common/HoodieSparkEngineContext.java

```diff
@@ -207,15 +206,15 @@ public void putCachedDataIds(HoodieDataCacheKey cacheKey, int... ids) {
   @Override
   public List<Integer> getCachedDataIds(HoodieDataCacheKey cacheKey) {
     synchronized (cachedRddIds) {
-      return cachedRddIds.getOrDefault(cacheKey, Collections.emptyList());
+      return cachedRddIds.getOrDefault(cacheKey, new ArrayList<>());
```

Comment:
why this change

Reply:
fix this error, empty list class can not add item, I will merge two lists
```shell
java.lang.UnsupportedOperationException
	at java.util.AbstractList.add(AbstractList.java:148)
	at java.util.AbstractList.add(AbstractList.java:108)
```
Another way, I can new a ArrayList to addAll items, just like:
```java
if (config.areReleaseResourceEnabled()) {
      HoodieSparkEngineContext sparkEngineContext = (HoodieSparkEngineContext) context;
      Map<Integer, JavaRDD<?>> allCachedRdds = sparkEngineContext.getJavaSparkContext().getPersistentRDDs();
      List<Integer> allDataIds = new ArrayList<>(sparkEngineContext.removeCachedDataIds(HoodieDataCacheKey.of(basePath, instantTime)));
      if (config.isMetadataTableEnabled()) {
        String metadataTableBasePath = HoodieTableMetadata.getMetadataTableBasePath(basePath);
        allDataIds.addAll(sparkEngineContext.removeCachedDataIds(HoodieDataCacheKey.of(metadataTableBasePath, instantTime)));
      }
      for (int id : allDataIds) {
        if (allCachedRdds.containsKey(id)) {
          allCachedRdds.get(id).unpersist();
        }
      }
    }
```

Reply:
> Another way, I can new a ArrayList to addAll items, just like:

This is always better.